refactor: replace uuid with native Crypto API

mag123c opened the pull request:
### Description of change
This change by removing unnecessary external cryptographic dependencies (sha.js and uuid) and replacing them with native Crypto API functionality available in modern Node.js and browser environments.


### Pull-Request Checklist

-   [x] Code is up-to-date with the `master` branch
-   [x] This pull request links relevant issues as Fixes #11693 
-   [x] There are new or updated unit tests validating the change
-   [ ] Documentation has been updated to reflect this change


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Chores**
  * Removed the external UUID dependency from the project.

* **Refactor**
  * Centralized UUID generation into an internal utility and replaced dispersed direct UUID usage with the new provider.
  * Implemented a cross-platform UUID v4 generator that uses native crypto where available and falls back to a deterministic method for broader reliability.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

alumni reviewed (changes requested):
Can break platforms that don't support crypto. Read the discussion in https://github.com/typeorm/typeorm/issues/11693

Also: the internal sha1 implementation has a flaw (when whitespace chars are in the input) - so this is a breaking change that cannot be merged into master (we use master for the 0.3 version).

mag123c commented:
@alumni
Thanks for the feedback! I'd appreciate your review to confirm this approach is correct.      

## Changes Made
**Platform Compatibility**
- Added multi-level fallback for UUID generation:
- Primary: `crypto.randomUUID()` (Node.js, modern browsers)
- Fallback 1: `crypto.getRandomValues()` (older browsers, React Native)
- Fallback 2: `Math.random()` (last resort)

Updated both Node and Browser PlatformTools implementations

**SHA-1 Breaking Change**
- **Approach taken**: Use `RandomGenerator.sha1()` on ALL platforms instead of native crypto
- **Reason**: Avoids breaking existing database constraint names
- **Trade-off**: Keeps the whitespace handling behavior (encodeURIComponent) for consistency

Please let me know if this direction is acceptable or if you'd prefer a different approach.

mag123c commented:
Fixed missing `socks@2.8.7` dependency in package-lock.json.

The entry was accidentally removed during a previous lock update when npm pruned the optional package. Manually restored from master's lock file.

alumni commented:
@pkuczynski I think here it's a problem of how we created the tasks, we should probably create 3 of them (shajs, uuid, and buffer), not just one :P

mag123c commented:
I've resolved the conflicts after the base branch was changed to `next`

pujux commented:
seems the lockfile needs an update so the checks run correctly :) @mag123c

mag123c commented:
@pujux Oh, thanks for mention.

Conflict was solved. @pkuczynski @alumni

alumni commented on src/util/RandomGenerator.ts:
```diff
@@ -1,4 +1,78 @@
+            } catch (e) {
+                // Fall through to custom implementation
+            }
+        }
```
Let's use more modern JS, if TS allows it. This applies in other places in this PR, but I'll only write it here.

```suggestion
        const uuid = globalThis.crypto?.randomUUID?.()
        if (uuid) {
           return uuid
        }
```

alumni commented on src/platform/PlatformTools.ts:
```diff
@@ -299,4 +287,13 @@ export class PlatformTools {
+     * Generates UUID v4 using native crypto API with fallback for environments
+     * that don't support it (e.g., React Native, Hermes).
+     */
+    static generateUuid(): string {
```
This would make more sense for SHA-1 which we already merged CC: @G0maa 

But for UUIDv4, since the implementation is generic, we probably don't need to do it, we can use the implementation directly, without having to go through `PlatformTools`.

G0maa reviewed (approved):
Few nitpicks, seems generally ok.

G0maa commented on src/util/RandomGenerator.ts:
```diff
@@ -1,4 +1,54 @@
+        const uuid = globalThis.crypto?.randomUUID?.()
+        if (uuid) {
+            return uuid
+        }
```
I see [your](https://github.com/typeorm/typeorm/pull/11769#discussion_r2657931168) comment references this above, so just to clarify: It seems in different places we have to do the same logic: `If node do x, if browser do y, if RN do z.` so I was suggesting a "standarized" way of doing this somewhere. i.e. such that every PR uses the same method.

alumni replied:
I'd keep it as it is - we don't need to do things based on which platform we're running, but based on what features the platform provides. So if the platform provides the `randomUUID` function, we should use it.

Over time we should get rid of `PlatformTools` if possible.

mag123c commented on src/util/RandomGenerator.ts:
```diff
@@ -1,4 +1,54 @@
+            // This approach is also used by expo-crypto and other RN libraries
+            // Note: For TypeORM's use case (DB IDs, cache IDs), uniqueness is sufficient
+            for (let i = 0; i < 16; i++) {
+                randomBytes[i] = Math.floor(Math.random() * 256)
```
Yes, for uniqueness it's sufficient. This fallback is only for RN/Hermes environments, and was already discussed with @alumni above

G0maa commented:
I think this PR is ready @alumni
